src/kubernetes_cluster/spec/pod_event/state_machine.rs

```diff
@@ -0,0 +1,118 @@
+// Copyright 2022 VMware, Inc.
+// SPDX-License-Identifier: MIT
+#![allow(unused_imports)]
+use super::types::*;
+use crate::external_api::spec::*;
+use crate::kubernetes_api_objects::spec::{pod::*, resource::*};
+use crate::kubernetes_cluster::spec::{cluster::Cluster, message::*};
+use crate::reconciler::spec::reconciler::Reconciler;
+use crate::state_machine::action::*;
+use crate::state_machine::state_machine::*;
+use crate::temporal_logic::defs::*;
+use vstd::{multiset::*, prelude::*};
+
+verus! {
+
+impl <K: CustomResourceView, E: ExternalAPI, R: Reconciler<K, E>> Cluster<K, E, R> {
+
+pub open spec fn create_pod() -> PodEventAction<E::Input, E::Output> {
+    Action {
+        precondition: |input: PodEventActionInput, s: PodEventState| {
+            &&& s.enabled
+            &&& input.obj.metadata.name.is_Some()
+            &&& input.obj.metadata.namespace.is_Some()
+            &&& input.obj.kind == PodView::kind()
+            &&& PodView::unmarshal(input.obj).is_Ok()
+        },
+        transition: |input: PodEventActionInput, s: PodEventState| {
+            let create_req_msg = Message::pod_event_req_msg(Message::create_req_msg_content(
+                input.obj.metadata.namespace.get_Some_0(),
+                input.obj,
+                input.rest_id_allocator.allocate().1
+            ));
+
+            let s_prime = s;
+            let output = PodEventActionOutput {
+                send: Multiset::singleton(create_req_msg),
+                rest_id_allocator: input.rest_id_allocator.allocate().0,
+            };
+            (s_prime, output)
+        },
+    }
+}
+
+pub open spec fn delete_pod() -> PodEventAction<E::Input, E::Output> {
+    Action {
+        precondition: |input: PodEventActionInput, s: PodEventState| {
+            &&& s.enabled
+            &&& input.obj.metadata.name.is_Some()
+            &&& input.obj.metadata.namespace.is_Some()
+            &&& input.obj.kind == PodView::kind()
+            &&& PodView::unmarshal(input.obj).is_Ok()
+        },
+        transition: |input: PodEventActionInput, s: PodEventState| {
+            let delete_req_msg = Message::pod_event_req_msg(Message::delete_req_msg_content(
+                input.obj.object_ref(), input.rest_id_allocator.allocate().1
+            ));
+
+            let s_prime = s;
+            let output = PodEventActionOutput {
+                send: Multiset::singleton(delete_req_msg),
+                rest_id_allocator: input.rest_id_allocator.allocate().0,
+            };
+            (s_prime, output)
+        },
+    }
+}
+
+// NOTE: We don't need to model update_status, since its effect can be modeled
```

Comment:
We need one for update_status: update won't change anything in the `status` field; only update_status does that.